client/components/Calendar.jsx

```diff
@@ -14,26 +14,63 @@ const Calendar = (props) => {
     [0, 0, 0, 0, 0, 0, 0],
     [0, 0, 0, 0, 0, 0, 0],
   ];
-  // get data for first day of current month
+
   const first = new Date(props.year, props.month, 1);
-  
```

Comment:
Would maybe suggest defining 'first' as 'monthStartDay' and just immediately accessing its day property, since that's the only way it's used here (unless my comment toward the end is mistaken, which it very well could be). Maybe also have 'last' be 'monthEndDate' (and defined as new Date(props.year, props.month + 1, 0).getDate()), since that is the only way it is being used in this component.